### PR TITLE
scalding

### PR DESCRIPTION
**Pull request by helenahm**
here is scalding code

### CODE REVIEW

File: scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import cascading.tuple.TupleEntry
+
+class Main ( args: Args ) extends Job( args )
+  {
+  
+    val inputFields = 'line
+	val users = ( 'id, 'email, 'language, 'location)
+  	val transactions = ( 'transaction_id, 'product_id, 'user_id, 'purchase_amount, 'item_description);
+	
+	val input1 = TextLine( args( "input1" ) )
```
Comment (rathboma) [inline scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala:12]: could you add the test files to the codebase so people can just run this and have it work without having to make them? Maybe put them somewhere like `data/users` and `data/transactions`?

File: scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	val input2 = TextLine( args( "input2" ) )
+	val output = Tsv( args( "output" ) )
+	
+	val usersInput = input1.read.mapTo( inputFields -> users )
+    { te: TupleEntry =>
```
Comment (rathboma) [inline scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala]: Scala coding convention is that open curly follows the first set of parens, like this:

```
val a = someFunction("hello") { world =>
  (world, "is awesome")
}
```

so this also means the code inside the closure is also indented more than the code surrounding it.

File: scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package main.scala.com.matthewrathbone.scalding
+
+import com.twitter.scalding._
+import cascading.tuple.TupleEntry
+
+class Main ( args: Args ) extends Job( args )
+  {
```
Comment (rathboma) [inline scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala]: see comment below on scala curly paren conventions.

File: scalding/src/test/scala/com/matthewrathbone/scalding/testMainFunctionJoinTest.scala [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+@RunWith(classOf[JUnitRunner])
+class MainFunctionJoinTest extends WordSpec with Matchers {
+  "Our job" should {
+    JobTest(new main.scala.com.matthewrathbone.scalding.Main(_))
+      .arg("input1", "usersInput")
+      .arg("input2", "transactionsInput")
+      .arg("output", "outputFile")
+      .source(TextLine("usersInput"), List((0, "1	matthew@test.com	EN	US"), (1, "2	matthew@test2.com	EN	GB"), (2, "3	matthew@test3.com	FR	FR")))
+      .source(TextLine("transactionsInput"), List((0, "1	1	1	300	a jumper"), (1, "2	1	2	300	a jumper"), (2, "3	1	2	300	a jumper"), (3, "4	2	3	100	a rubber chicken"), (4, "5	1	3	300	a jumper")))
+      .sink[(Int, Int)](Tsv("outputFile")){ outputBuffer =>
```
Comment (rathboma) [inline scalding/src/test/scala/com/matthewrathbone/scalding/testMainFunctionJoinTest.scala:19]: nice, yes indenting like this!

**Comment by rathboma** [comment]
This looks great overall, if we can address those minor issues I'll go ahead and merge it afterwards.

File: scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala (hunk 1 above)
Comment (helenahm) [inline scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala:12]: Hi Matthew, yes, I have added transactions_test.txt and users_test.txt to the scalding folder, in the same way as they live in the cascading folder, without data/ .

File: scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala (hunk 2 above)
Comment (helenahm) [inline scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala]: Changed.

File: scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala (hunk 3 above)
Comment (helenahm) [inline scalding/src/main/scala/com/matthewrathbone/scalding/Main.scala]: Changed.

**Pull request merged**